mlflow/__init__.py

```diff
@@ -50,76 +51,110 @@
     tracking,  # noqa: F401
 )
 
-# model flavors
-_model_flavors_supported = []
-try:
-    from mlflow import (
-        catboost,  # noqa: F401
-        diviner,  # noqa: F401
-        fastai,  # noqa: F401
-        gluon,  # noqa: F401
-        h2o,  # noqa: F401
-        johnsnowlabs,  # noqa: F401
-        langchain,  # noqa: F401
-        lightgbm,  # noqa: F401
-        llm,  # noqa: F401
-        mleap,  # noqa: F401
-        onnx,  # noqa: F401
-        openai,  # noqa: F401
-        paddle,  # noqa: F401
-        pmdarima,  # noqa: F401
-        prophet,  # noqa: F401
-        pyfunc,  # noqa: F401
-        pyspark,  # noqa: F401
-        pytorch,  # noqa: F401
-        recipes,  # noqa: F401
-        sentence_transformers,  # noqa: F401
-        shap,  # noqa: F401
-        sklearn,  # noqa: F401
-        spacy,  # noqa: F401
-        spark,  # noqa: F401
-        statsmodels,  # noqa: F401
-        tensorflow,  # noqa: F401
-        transformers,  # noqa: F401
-        xgboost,  # noqa: F401
-    )
-
-    _model_flavors_supported = [
-        "catboost",
-        "fastai",
-        "gluon",
-        "h2o",
-        "lightgbm",
-        "mleap",
-        "onnx",
-        "pyfunc",
-        "pytorch",
-        "sklearn",
-        "spacy",
-        "spark",
-        "statsmodels",
-        "tensorflow",
-        "keras",
-        "xgboost",
-        "shap",
-        "paddle",
-        "prophet",
-        "pmdarima",
-        "diviner",
-        "transformers",
-        "langchain",
-        "llm",
-        "openai",
-        "sentence_transformers",
-        "johnsnowlabs",
-    ]
-except ImportError:
-    # We are conditional loading these commands since the skinny client does
-    # not support them due to the pandas and numpy dependencies of MLflow Models
-    pass
+# Lazily load mlflow flavors to avoid excessive dependencies.
+catboost = LazyLoader("mlflow.catboost", globals(), "mlflow.catboost")
```

Comment:
Curious how this affects code autocomplete on Databricks notebook.

Reply:
On Databricks, we run code during the Python REPL setup. That may touch `mlflow.tesnroflow.xxx`.

Reply:
Good call, very likely.